### PR TITLE
chore: do not parse if text.length is more than 4000

### PR DESCRIPTION
**Pull request by dominictb**
### Details

https://github.com/Expensify/App/issues/46766#issuecomment-2459531934.

Here I lower down the text length by 4000, as at 5000 the iOS app is still a bit laggy

### Related Issues

https://github.com/Expensify/App/issues/46766

### CODE REVIEW

File: parser/index.ts [hunk 1]
```diff
@@ -273,6 +275,9 @@ function groupRanges(ranges: MarkdownRange[]) {
 
 function parseExpensiMarkToRanges(markdown: string): MarkdownRange[] {
   try {
+    if (markdown.length > MAX_PARSABLE_LENGTH) {
+      return [];
+    }
```
Comment (tomekzaw) [inline parser/index.ts:280]:
```suggestion
  if (markdown.length > MAX_PARSABLE_LENGTH) {
    return [];
  }
  try {
```

**Comment by tomekzaw** [comment]
Please regenerate `react-native-live-markdown-parser.js` to fix failing CI

**Review by tomekzaw — APPROVED** [review]
Thanks for the PR!

**Comment by d-haxton** [comment]
What an absolutely horrible development experience. You should _never_ just quietly fail. Add a log. Add it to the readme. This is so frustratingly bad.

**Comment by tomekzaw** [comment]
@d-haxton Sorry to hear that, I understand that this can be frustrating. Keep in mind that you can customize this and provide your own parser, `parseExpensiMark` is just a default.

**Pull request merged**